libyul/backends/wasm/WasmDialect.cpp

```diff
@@ -224,5 +236,5 @@ void WasmDialect::addFunction(
 	f.returns = std::move(_returns);
 	f.sideEffects = _movable ? SideEffects{} : SideEffects::worst();
 	f.isMSize = false;
-	f.literalArguments = _literalArguments;
+	f.literalArguments = std::move(_literalArguments);
```

Comment:
Please don't move a non-const ref.

Reply:
but... you can move _only_ a non-const ref?! At least according to: http://yacoder.guru/blog/2015/05/06/cpp-curiosities-one-does-not-simply-move-a-const-object/

I mean, a move is destructive, it needs to modify the object?

Reply:
If I make it a `const` parameter, `vector::operator=(const vector& __x)` is being used.
If i keep it like this, `vector::operator=(vector&& __x) noexcept(_Bit_alloc_traits::_S_nothrow_move())` is used..

Reply:
Best practice nowadays is to only move from a param that uses call-by-value, but you should at least pass by rvalue reference.